scripts/test_update_from_tag.sh

```diff
@@ -199,11 +199,24 @@ docker_pgcmd ${CONTAINER_UPDATED} "ALTER EXTENSION timescaledb UPDATE" "postgres
 if [[ "${TEST_VERSION}" > "v6" ]] || [[ "${TEST_VERSION}" = "v6" ]]; then
 	echo "Executing post update scritps"
 	docker_pgscript ${CONTAINER_UPDATED} /src/test/sql/updates/post.update.sql "single"
+	docker_pgscript ${CONTAINER_UPDATED} /src/test/sql/updates/post.repair.sql "single"
 fi
 
+# Check that there is nothing wrong before taking a backup
+echo "Checking that there are no missing dimension slices"
+docker_pgscript ${CONTAINER_UPDATED} /src/test/sql/updates/setup.check.sql
+
 echo "Executing setup script on clean"
 docker_pgscript ${CONTAINER_CLEAN_RERUN} /src/test/sql/updates/setup.${TEST_VERSION}.sql
 
+# We need to run the post repair script to make sure that the
+# constraint is on the clean rerun as well since the setup script can
+# remove it.
+if [[ "${TEST_VERSION}" > "v6" ]] || [[ "${TEST_VERSION}" = "v6" ]]; then
```

Comment:
Shouldn't the repair test be tied to specific TimescaleDB versions? I.e., we want to make sure that any TimescaleDB version that proceeds the repair update script will have the error, but no other. Is this handled by "v6"? I worry about the case when we also run repair tests for future upgrades where the issue no longer exists and the update script won't run (e.g., 2.0.1 -> 2.1.0)

Reply:
This check just make sure that we restore the constraint if it was dropped. The constraint is always dropped in v6 or later, hence needs to be restored in v6 and later. Inside the repair setup, there is a check that we are upgrading from a version before 2.0.0, which guarantees that the latest repair is done. If we are upgrading from 2.0.1 to 2.1.0, no dimension slices will be dropped, but the foreign key constraint still need to be restored.

This is because there is no easy way to check what we're upgrading from and to, only what version we are currently running as.

Reply:
Not sure I follow completely, but it sounds like the repair test activates only in cases we are upgrading from a pre 2.0 release.

One concern I still have is that we now do a separate CI run for the repair, which means we run all update tests from 2.0.1 to future releases multiple times, since the repair doesn't happen there. It seems somewhat unnecessary to run these update tests again. Or am I missing something?

Reply:
Yes, repair tests are only activated when upgrading from pre-2.0 to 2.0. The check has to be updated when merging with 1.7 branch since repair tests are only running on 1.7.1 to later.

No, you got it right. Right now we run all update tests with and without repair because there are repairs that happen during update. For branches where there is no repair, we need to disable these tests, but that is a later problem. For the time being, we will at least upgrade from 1.7.x branch, and there will always be a repair done until 1.7 EOL.